src/backend/api/Fusion.Resources.Domain/Commands/Departments/DeleteDelegatedResourceOwner.cs

```diff
@@ -0,0 +1,43 @@
+﻿using Fusion.Resources.Database;
+using MediatR;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Fusion.Resources.Domain.Commands.Departments
+{
+    public  class DeleteDelegatedResourceOwner : IRequest<bool>
+    {
+        private readonly string departmentId;
+        private readonly Guid delegatedOnwerAzureUniqueId;
+
+        public DeleteDelegatedResourceOwner(string departmentId, Guid delegatedOnwerAzureUniqueId)
+        {
+            this.departmentId = departmentId;
+            this.delegatedOnwerAzureUniqueId = delegatedOnwerAzureUniqueId;
+        }
+
+        public class Handler : IRequestHandler<DeleteDelegatedResourceOwner, bool>
+        {
+            private readonly ResourcesDbContext db;
+
+            public Handler(ResourcesDbContext db)
+            {
+                this.db = db;
+            }
+
+            public async Task<bool> Handle(DeleteDelegatedResourceOwner request, CancellationToken cancellationToken)
+            {
+                var query = db.DepartmentResponsibles
```

Comment:
I prefer to query for items, and remove(range) if found, then only call savechanges if actually modifying the database

Reply:
if(query.Any){removerange, savechanges}

Reply:
Wouldn't that require two roundtrips to the database? (I think the call to .Any() enumerates the enumerable). Whereas the method used here will only do one roundtrip to the database. Sending delete x where y when running savechangesasync().

Reply:
probably ok, just asking :-)